docs/dictionary-data.md

```diff
@@ -1,76 +1,120 @@
+keep separate the two distinct concepts of (1) word classes whose members
+are not declinable because that’s how the language works, and (2) word
+classes that should be declinable but where the required linguistic data
+together with a compatible computational model implementation is not
+presently at hand.
+
+## Dictionary entry level
+
+For each dictionary entry, we will need the following pieces of data:
+
+  - The entry head. In most dictionaries this is usually a single word, but
+    could also be a phrase or morpheme.
+
+  - The inflectional category. Plains Cree can use inflectional categories
+    such as `IPH` “indeclinable phrase” and `IPM` “indeclinable morpheme”
+    for non-word heads.
+
+  - The FST lemma. This is specified iff the inflectional class is
+    declinable. To be clear: the FST lemma is the thing that gets plugged
+    into a paradigm layout template to generate associated wordforms. For
+    example, if the head is `kinîminâwâw` with FST analysis
+    `nîmiw+V+AI+Ind+2Pl`, then the FST lemma is `nîmiw`.
+
+  - A list of definitions for the head. Each definition is typically a
+    string, and a list of citation abbreviations. There is intent to enrich
+    this data model by specifying things such as cross-references, notes,
+    and examples; specifying that is not in this document’s scope at the
+    moment.
+
+  - The homograph key. In order to have stable ways to refer to different
+    homographs in the database and in URLs, whenever two or more heads have
+    the same text, each entry with a head that is a homograph must provide
+    a homograph key that is distinct from every other wordform.
+
+    For example, for the Plains Cree homograph `yôtin`, the lexicographer
+    might set the keys for three wordforms to `yôtin1` and `yôtin2` and
+    `yôtin3`; or to `yôtin-na` and `yôtin-ni` and `yôtin-v`; or following
+    any other scheme they desire, so long as the assignments will remain
+    stable.
+
+    For convenience when the dictionary source has its own stable
+    identifiers, the homograph key may be set on every dictionary entry,
+    and will only be used on entries that are actually homographs.
+
+    Note that homographs may or may not be in the same inflectional
+    category.
+
+  - Optionally, a static paradigm name. Specifying this is only permitted
+    if the specific word class should not be declinable. For example, in
+    Plains Cree, ‘niya’ from the indeclinable inflectional category `PrA`
+    “animate pronouns” has static paradigm `personal-pronouns`.
+
+  - Optionally, the linguistic stem. This is displayed on the website for
+    humans to see, but is not accessed or used for any other purposes by
+    the dictionary code. This is technically optional because not all
+    dictionary sources and language pairs will have this information and/or
+    want to display it.
+
+    For Plains Cree specifically: This is the `\stm` field in the *Cree:
```

Comment:
I would formulate this rather as follows, referring to the ALTLab crk-db, which is the immediate source of importation of the dictionary content (rather than CW content, which is processed to a certain extent when integrated into crk-db, and which contains currently certain information that other dictionaries such as MD and AECD do not include).

For Plains Cree specifically, there are two variants of linguistic stems in the ALTLab crk-db. For both, a preceding hyphen (for dependent nouns, e.g. _-ohkom-_) and following hyphen (for all stems, e.g. _nimî-_) indicate that they are not free morphemes (independent wordforms):
- the plain/minimal `stem` according to the `\stm` field in the CW toolbox source, which N.B. is lacking from MD and AECD. This minimal stem may lack reduplicative elements and/or preverbs/prenouns, and thus may not have a one-to-one mapping to possible lemmas (e.g. _api-_ as the minimal stem of _ay-apiw_.
- the full `FST-stem` according to the `fststem` field in the ALTLab crk-db. This includes all the reduplicative elements as well as preverbs/prenouns, and thus has a one-to-one mapping with the lemma. This is created in the ALTLab crk-db based on the plain/minimal stem field, e.g. _ay-api-_ as the full FST-stem for the lemma _ay-apiw_. The FST-stem, when supplemented with special morphophonological symbols, is used in the lexc source code for crk stems in the format: `&lt;FST lemma&gt;:&lt;FST stem&gt;`, e.g.,`acitakotêw:acitakot3 VTAt
- itwêwina may opt to show either the plain/minimal stem or the full FST-stem (or both). The code for generating the previously used XML crkeng source opted to pass the FST-stem to itwêwina.